### PR TITLE
Add 'wellgraph' utility program.

### PR DESCRIPTION
**Pull request by atgeirr**
This program reads a deck and creates a graph from the group/well structure. It is stored as a file in the "dot" language used by GraphViz.

For example, running it on a case from `opm-tests/model4`:
```
wellgraph MOD4_GRP.DATA
```
will result in the output file `MOD4_GRP.gv` containing:
```
// This file was written by the 'wellgraph' utility from OPM.
// Find the source code at github.com/OPM.
// Convert output to PDF with 'dot -Tpdf MOD4_GRP.gv > MOD4_GRP.pdf'
strict digraph MOD4_GRP
{
    FIELD -> { MOD4 }
    MOD4 -> { UPPER LOWER AQF }
    UPPER -> { MAIN SE }
    LOWER -> { CENTRAL }
    MAIN -> { NE NW }
    node [shape=box]
    AQF -> { WID1 WID2 }
    NE -> { OPU4 WIU3 OPU5 }
    NW -> { OPU1 OPU2 WIU1 OPU3 WIU2 }
    SE -> { OPU6 WIU4 OPU7 }
    CENTRAL -> { OPL1 OPL2 OPL3 WIL1 WIL2 OPL4 OPL5 }
    OPL1 [color=red]
    WID1 [color=blue]
    WID2 [color=blue]
    OPL2 [color=red]
    OPL3 [color=red]
    WIL1 [color=blue]
    WIL2 [color=blue]
    OPL4 [color=red]
    OPU1 [color=red]
    OPU2 [color=red]
    WIU1 [color=blue]
    OPU4 [color=red]
    WIU3 [color=blue]
    OPU6 [color=red]
    WIU4 [color=blue]
    OPL5 [color=red]
    OPU3 [color=red]
    WIU2 [color=blue]
    OPU5 [color=red]
    OPU7 [color=red]
}
```
Converting the graph to PDF using the dot command printed in the file comment yields a pdf.
[MOD4_GRP.pdf](https://github.com/OPM/opm-common/files/4532504/MOD4_GRP.pdf)

### CODE REVIEW

**Comment by atgeirr** [comment]
jenkins build this please

File: examples/wellgraph.cpp [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+int main(int argc, char** argv)
+{
+    std::ostringstream os;
+    std::shared_ptr<Opm::StreamLog> string_log = std::make_shared<Opm::StreamLog>(os, Opm::Log::DefaultMessageTypes);
+    Opm::OpmLog::addBackend( "STRING" , string_log);
+    try {
+        for (int iarg = 1; iarg < argc; iarg++) {
+            const std::string filename = argv[iarg];
+            const auto sched = loadSchedule(filename);
+            const auto casename = std::filesystem::path(filename).stem();
```
Comment (joakim-hove) [inline examples/wellgraph.cpp:124]: I guess you should use the `Opm::filesystem` shim:
```C++
#include <opm/common/utility/FileSystem.hpp>
...
const auto casename = Opm::filesystem::path(filename).stem();
```

**Comment by joakim-hove** [comment]
Nice addition - merge away!

**Comment by joakim-hove** [comment]
An idea: Let the arrows point in the stream direction - i.e. different from producers and injectors?

File: examples/wellgraph.cpp (hunk 1 above)
Comment (atgeirr) [reply examples/wellgraph.cpp]: Done

**Comment by atgeirr** [comment]
> An idea: Let the arrows point in the stream direction - i.e. different from producers and injectors?

Right now, the arrows indicate a parent->child relation, but since vertical position can do that this is a good idea. One may need to specify "node rank" in the dot output to pull this off perhaps. To determine production/injection status for connections between groups needs a bit of work, I suggest it is left for the next person to get inspired...

**Comment by atgeirr** [comment]
jenkins build this please

**Comment by bska** [comment]
Very cool.  A possible future extension might be to have the ability to draw/output all configurations rather than just the final&mdash;at least if the group tree itself changes during the run&mdash;and to possibly have separate graphs for the group-to-well connections.  The latter will typically be more numerous than the group-to-group connections.

In any case, I'll merge this now.  Thanks a lot for the new feature.

**Pull request merged**